transport/conn_integration_test.go

```diff
@@ -160,3 +160,89 @@ func TestCloseHangingIntegration(t *testing.T) {
 
 	wg.Wait()
 }
+
+func newConnCompressionTestHelper(t testing.TB, c frame.Compression) *connTestHelper {
+	cfg := ConnConfig{
+		Timeout:     500 * time.Millisecond,
+		Compression: c,
+	}
+	conn, err := OpenConn(TestHost+":9042", nil, cfg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return &connTestHelper{t: t, conn: conn}
+}
+
+func TestSnappyCompressionIntegration(t *testing.T) {
+	h := newConnCompressionTestHelper(t, frame.Snappy)
+	h.applyFixture()
+
+	defer h.close()
```

Comment:
It would make more sense to refactor all the test code to a helper function and call that function in a test (named subtest maybe) with compression as a parameter.